backend/src/Edna/Web/Types.hs

```diff
@@ -17,4 +43,180 @@ data ExperimentalMeasurement = ExperimentalMeasurement
   , emOutlier :: Bool
   } deriving stock (Generic, Show, Eq)
 
-deriveJSON ednaAesonWebOptions ''ExperimentalMeasurement
+----------------
+-- General types
+----------------
+
+-- | A simple wrapper over 'Word'. At the data layer, we identify all entities
+-- with numeric IDs and this data type corresponds to such an ID.
+-- It has a phantom parameter type which the type of the object
+-- identified by this ID.
+newtype SqlId t = SqlId
+  { unSqlId :: Word
+  } deriving stock (Generic, Show)
+    deriving newtype (FromHttpApiData, FromJSON, ToJSON, ToSchema)
+
+-- | This data type is useful when you want to return something with its ID.
+data WithId t = WithId
+  { wiId :: SqlId t
+  , wItem :: t
+  } deriving stock (Generic, Show)
+
+-- | This data type is used when you want to return something with its ID and
+-- some additional data that was not submitted by end users, but is maintained
+-- by the application.
+data WithExtra t e = WithExtra
+  { weId :: SqlId t
+  , weItem :: t
+  , weExtra :: e
+  } deriving stock (Generic, Show)
+
+-- | A stub to specify the sorting order, most likely will be replaced with
+-- @servant-util@.
+data StubSortBy =
+    SortByName
+  | SortBySomething
+
+instance FromHttpApiData StubSortBy where
+  parseQueryParam = \case
+    "name" -> pure SortByName
+    "something" -> pure SortBySomething
+    x -> Left $ "unknown sorting order: " <> x
+
+----------------
+-- Entities
+----------------
+
+-- | Input data submitted along with uploaded file.
+data FileUploadInput = FileUploadInput
+  { fuiProject :: SqlId Project
+  -- ^ ID of the project the file belongs to.
+  , fuiTestMethodology :: SqlId TestMethodology
+  -- ^ ID of the test methodology used throughout the file.
+  } deriving stock (Generic, Show)
+
+-- | Summary of an experiment data file.
+newtype FileSummary = FileSummary
+  { unFileSummary :: [FileSummaryItem]
+  } deriving stock (Generic, Show)
+
+-- | One element in 'FileSummary'. Corresponds to one target from the file.
+-- Contains all compounds that interact with the target in the file.
+-- Also contains information whether this target is new or already known.
+data FileSummaryItem = FileSummaryItem
+  { fsiTarget :: SqlId Target
+  -- ID of a target from the file.
+  , fsiCompounds :: [SqlId Compound]
+  -- IDs of all compounds interacting with this target.
+  , fsiIsNew :: Bool
+  -- Whether this target is new, i. e. wasn't uploaded before.
+  } deriving stock (Generic, Show)
```

Comment:
Can you please explain the logic of this params and corresponding (it is duplicate of the comment above, but I want to explain my opinion). First of all what for we need to separate `parse` and `upload` entry points. Secondly I don't understand why here we do not return points and return only information about compounds and targets.

Reply:
Okay, now I understand why we separate them, but question about points is still interests me

Reply:
On the Upload page we have a preview table which summarizes the uploaded file. This is how it looks:
![image](https://user-images.githubusercontent.com/2556409/111277939-0d134280-864a-11eb-8f25-c9bc85e65b7b.png)

So `FileSummary` directly corresponds to that table. There is no place for points, it's a summary which lets one quickly check which compounds and targets will be added.

Reply:
Wait, and what is under the view button?

Reply:
`View` is activated only after the file is added. When you click it, you go to Dashboard.

Reply:
By "activated" I mean that you can click on it.